extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -405,12 +422,19 @@ private void applyConfig(Session session, Project project, String target, String
      * This case comes up when users want to deploy their application to a cluster like Minikube where no registry is used
      * and instead they rely on the image being built directly into the docker daemon that the cluster uses.
      */
-    private boolean needToForceUpdateImagePullPolicy(Optional<ContainerImageInfoBuildItem> containerImage,
+    private boolean needToForceUpdateImagePullPolicy(Collection<String> deploymentTargets,
+            Optional<ContainerImageInfoBuildItem> containerImage,
             Capabilities capabilities) {
+
+        // no need to change when we use Minikube only
+        if ((deploymentTargets.size() == 1) && MINIKUBE.equals(deploymentTargets.iterator().next())) {
```

Comment:
`... && deploymentTargets.contains(MINIKUBE)` should work

Reply:
Right, it's left over from what I was doing in an earlier iteration